### PR TITLE
Make allowHost, allowReflexive and allowRelay configurable

### LINKED ISSUE #3: Options: allow setting allowHost, allowReflexive, allowRelay via IceOptions
See [here](https://github.com/FAForever/kotlin-ice-adapter/blob/530f6fc4fe98ebafbd8d1b58dade1ef4d1eadae2/kia-lib/src/main/kotlin/com/faforever/ice/ice4j/AgentWrapper.kt#L111-L113)

### PR DESCRIPTION
**Pull request by BlackYps**
closes #3 

I didn't know what to put into the parameter desctriptions.
Also I don't know if localPlayerId was supposed to be refactored in this way, but it made sense to me.

### CODE REVIEW

File: kia-cli/src/main/kotlin/com/faforever/ice/KiaApplication.kt [hunk 1]
```diff
@@ -53,8 +53,17 @@ class KiaApplication : Callable<Int> {
...
+
+    @Option(names = ["--allow-reflexive"], description = ["???"])
+    private var allowReflexive: Boolean = true
+
+    @Option(names = ["--allow-relay"], description = ["???"])
+    private var allowRelay: Boolean = true
+
     override fun call(): Int {
-        val iceOptions = IceOptions(userId, userName, gameId, forceRelay, lobbyPort, gpgnetPort, telemetryServer)
+        val iceOptions = IceOptions(userId, userName, gameId, forceRelay, lobbyPort, gpgnetPort, telemetryServer, allowHost, allowReflexive, allowRelay)
```
Comment (Brutus5000) [inline kia-cli/src/main/kotlin/com/faforever/ice/KiaApplication.kt]: The cli args should stay compatible with the current ICE adapter for now. I think there is a force relay option, which should these 3 options accordingly. (You'll need to lookup the old code)

**Comment by Brutus5000** [comment]
To answer your question: I'm not so super happy with passing the iceOptions around everywhere, but until the core functionality is all working, there isn't much point in not doing it as long as we don't really know what is needed where. Maybe it makes sense to split the interface into sub-interfaces later on.

**Comment by BlackYps** [comment]
what interface and subinterfaces are you talking about?

File: kia-cli/src/main/kotlin/com/faforever/ice/KiaApplication.kt (hunk 1 above)
Comment (BlackYps) [reply kia-cli/src/main/kotlin/com/faforever/ice/KiaApplication.kt]: uhh, I think you are missing an improvent verb here

Comment (BlackYps) [reply kia-cli/src/main/kotlin/com/faforever/ice/KiaApplication.kt]: Oh, I see the java adapter just sets these three depending on the force relay option

**Comment by Brutus5000** [comment]
I was talking about IceOptions, but i see it's a data class. I was thinking of interface from the other Quarkus projects :D 
So the IceOptions data class could implement multiple interfaces that just show a "view" to some codependent variables.
But that is not relevant right now.

File: kia-lib/src/main/kotlin/com/faforever/ice/ice4j/AgentWrapper.kt [hunk 2]
```diff
@@ -103,14 +104,21 @@ class AgentWrapper(
                     setState(IceState.AWAITING_CANDIDATES)
                 }
 
+                var allowHost = true
+                var allowReflexive = true
+                if (forceRelay) {
+                    allowHost = false
+                    allowReflexive = false
+                }
```
Comment (Brutus5000) [inline kia-lib/src/main/kotlin/com/faforever/ice/ice4j/AgentWrapper.kt]: In Kotlin avoid `var` if possible. if/else is an expression, so you could say
`val bla = if(somethinge) 1 else -1`

In this case `val allowHost = !forceRelay` and `val allowReflex = !forceRelay` seems better though.

Comment (Brutus5000) [reply kia-lib/src/main/kotlin/com/faforever/ice/ice4j/AgentWrapper.kt]: Or you pass `!forceRelay` directly as argument in the method call

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**